kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Schemas.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spi;
+
+import com.google.common.io.Closeables;
+import com.google.common.io.Resources;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import parquet.avro.AvroSchemaConverter;
+import parquet.hadoop.ParquetFileReader;
+import parquet.hadoop.metadata.ParquetMetadata;
+
+public class Schemas {
```

Comment:
No need to make this public - just leave it in the SPI for now.
On 30 Mar 2015 20:16, "Ryan Blue" <notifications@github.com> wrote:

> In
> kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Schemas.java
> <https://github.com/kite-sdk/kite/pull/321#discussion_r27422407>:
>
> > +import java.io.File;
> > +import java.io.IOException;
> > +import java.io.InputStream;
> > +import java.net.URI;
> > +import org.apache.avro.Schema;
> > +import org.apache.avro.file.DataFileStream;
> > +import org.apache.avro.generic.GenericDatumReader;
> > +import org.apache.avro.generic.GenericRecord;
> > +import org.apache.hadoop.conf.Configuration;
> > +import org.apache.hadoop.fs.FileSystem;
> > +import org.apache.hadoop.fs.Path;
> > +import parquet.avro.AvroSchemaConverter;
> > +import parquet.hadoop.ParquetFileReader;
> > +import parquet.hadoop.metadata.ParquetMetadata;
> > +
> > +public class Schemas {
>
> If this is moved to the public API as a follow-up, then I think Schemas
> is the right name for it. Otherwise you're right and we should probably
> rename it or move it. What do you think about making this public?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kite-sdk/kite/pull/321/files#r27422407>.
>